flink-runtime/src/test/java/org/apache/flink/runtime/state/TaskLocalStateStoreImplTest.java

```diff
@@ -59,12 +59,18 @@ public void before() throws Exception {
 		this.subtaskIdx = 0;
 		this.rootDirs = new File[]{temporaryFolder.newFolder(), temporaryFolder.newFolder()};
 
+		LocalRecoveryDirectoryProviderImpl directoryProvider =
+			new LocalRecoveryDirectoryProviderImpl(rootDirs, jobID, allocationID, jobVertexID, subtaskIdx);
+
+		LocalRecoveryConfig localRecoveryConfig =
+			new LocalRecoveryConfig(LocalRecoveryConfig.LocalRecoveryMode.DISABLED, directoryProvider);
```

Comment:
It's a bit strange that we have to pass in a `directoryProvider`, even though the `LocalRecoveryMode` is disabled. I think it would be better to have something like `LocalRecoveryConfig.disabled()` and `LocalRecoveryConfig.fileBased(directoryProvider)`.

Reply:
So wouldn't that mean that with `LocalRecoveryConfig.disabled()`, the directory provider is again nullable? I think that `LocalRecoveryDirectoryProvider` is a very lightweight object, that also always has a useful default configuration.